### DISCUSSION TITLE
Can markdown be passed in json as post request

### DISCUSSION THREAD

**Discussion by venumanthena** [remarkjs/remark#809] [Q&A] [answered]
Hi,
Can markdown be passed in json as post request?  As it contains characters that break the json string, what would be the preferred method to send markdown through an api.
 
Also usage of the below markdown which translates to a table using remark-gfm is breaking the json string.

 | Feature    | Support              |
| ---------: | :------------------- |
| CommonMark | 100%                 |
| GFM        | 100% w/ `remark-gfm` |

Let me know if any library exists which converts this to a proper string then parses it to correct markdown.

Thanks in advance.

Venu

**Comment by wooorm** [comment, marked as answer]
Yes, you can post markdown, either as a string or as an AST, over the wire in a POST request.
Here’s an example of how `fetch` works, if you’re working in a browser: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#supplying_request_options.
If you’re in Node, then [`node-fetch`](https://github.com/node-fetch/node-fetch) and [`undici`](https://github.com/nodejs/undici) are nice HTTP clients.

**Reply by wooorm** [reply]
Presumably you are sending this as a string. Not as a JSON string.
This question has nothing to do with markdown or remark, but that you’re sending invalid JSON. How to fix that depends on what tools you are using to send JSON.
